### PR TITLE
Add package source to "installing <package> <version>" message at normal verbosity

### LINKED ISSUE #10461: When restoring at normal verbosity, log which source a package is being restored from
Currently, when restoring at normal verbosity, I get an error message when a package is being installed, but it doesn't say from where. It might be implied if it's downloaded from an HTTP source, but if it's found on a local source, there's nothing in the log to explain where it came from.

Currently:

```text
Restoring packages for C:\src\NuGet.Client\build\bootstrap.proj...
Installing Newtonsoft.Json 9.0.1
Committing restore...
Writing assets file to disk. Path: C:\src\NuGet.Client\packages\obj\project.assets.json
Restored C:\src\NuGet.Client\build\bootstrap.proj (in 679 ms).
```

Desired: the "installing" line should read:

```text
Installing Newtonsoft.Json 9.0.1 from C:\src\nupkgs\.
```

Note, https://github.com/NuGet/NuGet.Client/pull/3700 https://github.com/NuGet/Home/issues/3055 added the source to diagnostic verbosity, but this request is for "normal" verbosity.

### PR DESCRIPTION
**Pull request by zivkan**
## Bug

Fixes: https://github.com/NuGet/Home/issues/10461
Regression: No  
* Last working version:   
* How are we preventing it in future:   

## Fix

Details: Add "from {source}" to the existing "installing {id} {version}" message.

## Testing/Validation

Tests Added: Yes
Reason for not adding tests:  
Validation:

### CODE REVIEW

File: src/NuGet.Core/NuGet.Packaging.Extraction/NuGet.Packaging.Extraction.csproj [hunk 1]
```diff
@@ -291,6 +291,7 @@
     <Compile Include="..\NuGet.Packaging\PackagingConstants.cs" Link="NuGet.Packaging\PackagingConstants.cs" />
     <Compile Include="..\NuGet.Packaging\VersionFolderPathResolver.cs" Link="NuGet.Packaging\VersionFolderPathResolver.cs" />
     <Compile Include="..\NuGet.Packaging\Sha512HashFunction.cs" Link="NuGet.Packaging\Sha512HashFunction.cs" />
+    <Compile Include="..\NuGet.Packaging\StringFormatter.cs" Link="SNuGetPackaging\StringFormatter.cs" />
```
Comment (nkolev92) [inline src/NuGet.Core/NuGet.Packaging.Extraction/NuGet.Packaging.Extraction.csproj]: what's the S here for?

File: src/NuGet.Core/NuGet.Packaging/PackageExtractor.cs [hunk 2]
```diff
@@ -398,11 +398,7 @@ public static class PackageExtractor
                             logger.LogVerbose(
                                 $"Acquired lock for the installation of {packageIdentity.Id} {packageIdentity.Version}");
 
-                            logger.LogInformation(string.Format(
-                                CultureInfo.CurrentCulture,
-                                Strings.Log_InstallingPackage,
-                                packageIdentity.Id,
-                                packageIdentity.Version));
+                            logger.LogInformation(StringFormatter.Log_InstallingPackage(packageIdentity.Id, packageIdentity.Version.OriginalVersion, source));
```
Comment (nkolev92) [inline src/NuGet.Core/NuGet.Packaging/PackageExtractor.cs]: Can you post the verbose logs of an installation?

**Review by nkolev92 — COMMENTED** [review]
Looks good, but is the debug message now redundant?

**Comment by zivkan** [comment]
## Sample outputs

### normal verbosity

<details><summary>With this PR</summary>

```text
MSBuild auto-detection: using msbuild version '16.9.0.57004' from 'C:\Program Files (x86)\Microsoft Visual Studio\2019\Preview\MSBuild\Current\bin'.
Restoring packages for C:\src\NuGet.Client\build\bootstrap.proj...
  GET https://api.nuget.org/v3-flatcontainer/newtonsoft.json/index.json
  GET https://dotnetfeed.blob.core.windows.net/dotnet-core/flatcontainer/newtonsoft.json/index.json
  OK https://dotnetfeed.blob.core.windows.net/dotnet-core/flatcontainer/newtonsoft.json/index.json 14ms
  OK https://api.nuget.org/v3-flatcontainer/newtonsoft.json/index.json 59ms
  GET https://pkgs.dev.azure.com/azure-public/3ccf6661-f8ce-4e8a-bb2e-eff943ddd3c7/_packaging/78665e4c-b767-412b-9804-2b1ef7a48b8a/nuget/v3/flat2/newtonsoft.json/index.json
  GET https://dotnetfeed.blob.core.windows.net/dotnet-core/flatcontainer/newtonsoft.json/9.0.1/newtonsoft.json.9.0.1.nupkg
  GET https://pkgs.dev.azure.com/azure-public/3ccf6661-f8ce-4e8a-bb2e-eff943ddd3c7/_packaging/2a239fd0-3e21-40b0-b9d6-bc122fec7eb2/nuget/v3/flat2/newtonsoft.json/index.json
  OK https://dotnetfeed.blob.core.windows.net/dotnet-core/flatcontainer/newtonsoft.json/9.0.1/newtonsoft.json.9.0.1.nupkg 40ms
  NotFound https://pkgs.dev.azure.com/azure-public/3ccf6661-f8ce-4e8a-bb2e-eff943ddd3c7/_packaging/2a239fd0-3e21-40b0-b9d6-bc122fec7eb2/nuget/v3/flat2/newtonsoft.json/index.json 28ms
  NotFound https://pkgs.dev.azure.com/azure-public/3ccf6661-f8ce-4e8a-bb2e-eff943ddd3c7/_packaging/78665e4c-b767-412b-9804-2b1ef7a48b8a/nuget/v3/flat2/newtonsoft.json/index.json 77ms
  GET https://pkgs.dev.azure.com/dnceng/9ee6d478-d288-47f7-aacc-f6e6d082ae6d/_packaging/1a5f89f6-d8da-4080-b15f-242650c914a8/nuget/v3/flat2/newtonsoft.json/index.json
  NotFound https://pkgs.dev.azure.com/dnceng/9ee6d478-d288-47f7-aacc-f6e6d082ae6d/_packaging/1a5f89f6-d8da-4080-b15f-242650c914a8/nuget/v3/flat2/newtonsoft.json/index.json 66ms
  GET https://pkgs.dev.azure.com/dnceng/9ee6d478-d288-47f7-aacc-f6e6d082ae6d/_packaging/40ffd17d-412e-4e95-b1ca-c5e6b3774cd2@dc1a9d3d-0b54-4674-a4c3-0f6e4269f5f4/nuget/v3/flat2/newtonsoft.json/index.json
  GET https://pkgs.dev.azure.com/dnceng/9ee6d478-d288-47f7-aacc-f6e6d082ae6d/_packaging/9d15d80a-6afc-4f7e-901b-9378146a4b8b/nuget/v3/flat2/newtonsoft.json/index.json
  NotFound https://pkgs.dev.azure.com/dnceng/9ee6d478-d288-47f7-aacc-f6e6d082ae6d/_packaging/40ffd17d-412e-4e95-b1ca-c5e6b3774cd2@dc1a9d3d-0b54-4674-a4c3-0f6e4269f5f4/nuget/v3/flat2/newtonsoft.json/index.json 128ms
  NotFound https://pkgs.dev.azure.com/dnceng/9ee6d478-d288-47f7-aacc-f6e6d082ae6d/_packaging/9d15d80a-6afc-4f7e-901b-9378146a4b8b/nuget/v3/flat2/newtonsoft.json/index.json 64ms
Installed Newtonsoft.Json 9.0.1 from https://dotnetfeed.blob.core.windows.net/dotnet-core/index.json with content hash U82mHQSKaIk+lpSVCbWYKNavmNH1i5xrExDEquU1i6I5pV6UMOqRnJRSlKO3cMPfcpp0RgDY+8jUXHdQ4IfXvw==.
Committing restore...
Assets file has not changed. Skipping assets file writing. Path: C:\src\NuGet.Client\packages\obj\project.assets.json
Restored C:\src\NuGet.Client\build\bootstrap.proj (in 1.15 sec).

NuGet Config files used:
    C:\src\NuGet.Client\NuGet.Config
    C:\Users\zivkan\AppData\Roaming\NuGet\NuGet.Config
    C:\Program Files (x86)\NuGet\Config\Microsoft.VisualStudio.FallbackLocation.config
    C:\Program Files (x86)\NuGet\Config\Microsoft.VisualStudio.Offline.config

Feeds used:
    https://api.nuget.org/v3/index.json
    https://dotnetfeed.blob.core.windows.net/dotnet-core/index.json
    https://pkgs.dev.azure.com/azure-public/vside/_packaging/vs-impl/nuget/v3/index.json
    https://pkgs.dev.azure.com/azure-public/vside/_packaging/vssdk/nuget/v3/index.json
    https://pkgs.dev.azure.com/dnceng/public/_packaging/dotnet-eng/nuget/v3/index.json
    https://pkgs.dev.azure.com/dnceng/public/_packaging/myget-legacy%40Local/nuget/v3/index.json
    https://pkgs.dev.azure.com/dnceng/public/_packaging/nuget-build/nuget/v3/index.json
```

</details>

<details><summary>NuGet 5.8</summary>

```text
MSBuild auto-detection: using msbuild version '16.9.0.57004' from 'C:\Program Files (x86)\Microsoft Visual Studio\2019\Preview\MSBuild\Current\bin'.
Restoring packages for C:\src\NuGet.Client\build\bootstrap.proj...
  CACHE https://api.nuget.org/v3-flatcontainer/newtonsoft.json/index.json
  CACHE https://dotnetfeed.blob.core.windows.net/dotnet-core/flatcontainer/newtonsoft.json/index.json
  GET https://pkgs.dev.azure.com/azure-public/3ccf6661-f8ce-4e8a-bb2e-eff943ddd3c7/_packaging/78665e4c-b767-412b-9804-2b1ef7a48b8a/nuget/v3/flat2/newtonsoft.json/index.json
  GET https://pkgs.dev.azure.com/azure-public/3ccf6661-f8ce-4e8a-bb2e-eff943ddd3c7/_packaging/2a239fd0-3e21-40b0-b9d6-bc122fec7eb2/nuget/v3/flat2/newtonsoft.json/index.json
  GET https://pkgs.dev.azure.com/dnceng/9ee6d478-d288-47f7-aacc-f6e6d082ae6d/_packaging/1a5f89f6-d8da-4080-b15f-242650c914a8/nuget/v3/flat2/newtonsoft.json/index.json
  GET https://pkgs.dev.azure.com/dnceng/9ee6d478-d288-47f7-aacc-f6e6d082ae6d/_packaging/40ffd17d-412e-4e95-b1ca-c5e6b3774cd2@dc1a9d3d-0b54-4674-a4c3-0f6e4269f5f4/nuget/v3/flat2/newtonsoft.json/index.json
  GET https://pkgs.dev.azure.com/dnceng/9ee6d478-d288-47f7-aacc-f6e6d082ae6d/_packaging/9d15d80a-6afc-4f7e-901b-9378146a4b8b/nuget/v3/flat2/newtonsoft.json/index.json
  NotFound https://pkgs.dev.azure.com/azure-public/3ccf6661-f8ce-4e8a-bb2e-eff943ddd3c7/_packaging/2a239fd0-3e21-40b0-b9d6-bc122fec7eb2/nuget/v3/flat2/newtonsoft.json/index.json 45ms
  NotFound https://pkgs.dev.azure.com/azure-public/3ccf6661-f8ce-4e8a-bb2e-eff943ddd3c7/_packaging/78665e4c-b767-412b-9804-2b1ef7a48b8a/nuget/v3/flat2/newtonsoft.json/index.json 79ms
Installing Newtonsoft.Json 9.0.1.
  GET https://api.nuget.org/v3-flatcontainer/newtonsoft.json/9.0.1/newtonsoft.json.9.0.1.nupkg
  NotFound https://pkgs.dev.azure.com/dnceng/9ee6d478-d288-47f7-aacc-f6e6d082ae6d/_packaging/1a5f89f6-d8da-4080-b15f-242650c914a8/nuget/v3/flat2/newtonsoft.json/index.json 78ms
  OK https://api.nuget.org/v3-flatcontainer/newtonsoft.json/9.0.1/newtonsoft.json.9.0.1.nupkg 18ms
  NotFound https://pkgs.dev.azure.com/dnceng/9ee6d478-d288-47f7-aacc-f6e6d082ae6d/_packaging/40ffd17d-412e-4e95-b1ca-c5e6b3774cd2@dc1a9d3d-0b54-4674-a4c3-0f6e4269f5f4/nuget/v3/flat2/newtonsoft.json/index.json 74ms
  NotFound https://pkgs.dev.azure.com/dnceng/9ee6d478-d288-47f7-aacc-f6e6d082ae6d/_packaging/9d15d80a-6afc-4f7e-901b-9378146a4b8b/nuget/v3/flat2/newtonsoft.json/index.json 74ms
Committing restore...
Generating MSBuild file C:\src\NuGet.Client\build\../packages/obj/bootstrap.proj.nuget.g.props.
Assets file has not changed. Skipping assets file writing. Path: C:\src\NuGet.Client\packages\obj\project.assets.json
Restored C:\src\NuGet.Client\build\bootstrap.proj (in 1.32 sec).

NuGet Config files used:
    C:\src\NuGet.Client\NuGet.Config
    C:\Users\zivkan\AppData\Roaming\NuGet\NuGet.Config
    C:\Program Files (x86)\NuGet\Config\Microsoft.VisualStudio.FallbackLocation.config
    C:\Program Files (x86)\NuGet\Config\Microsoft.VisualStudio.Offline.config

Feeds used:
    https://api.nuget.org/v3/index.json
    https://dotnetfeed.blob.core.windows.net/dotnet-core/index.json
    https://pkgs.dev.azure.com/azure-public/vside/_packaging/vs-impl/nuget/v3/index.json
    https://pkgs.dev.azure.com/azure-public/vside/_packaging/vssdk/nuget/v3/index.json
    https://pkgs.dev.azure.com/dnceng/public/_packaging/dotnet-eng/nuget/v3/index.json
    https://pkgs.dev.azure.com/dnceng/public/_packaging/myget-legacy%40Local/nuget/v3/index.json
    https://pkgs.dev.azure.com/dnceng/public/_packaging/nuget-build/nuget/v3/index.json
```

</details>

Just installing/installed messages:

NuGet 5.8:

```text
Installing Newtonsoft.Json 9.0.1.
```

With this PR:
```text
Installed Newtonsoft.Json 9.0.1 from https://dotnetfeed.blob.core.windows.net/dotnet-core/index.json with content hash U82mHQSKaIk+lpSVCbWYKNavmNH1i5xrExDEquU1i6I5pV6UMOqRnJRSlKO3cMPfcpp0RgDY+8jUXHdQ4IfXvw==.
```

### detailed verbosity

<details><summary>With this PR</summary>

```text
NuGet Version: 5.9.0.45451
MSBuild P2P timeout [ms]: 120000
MSBuild auto-detection: using msbuild version '16.9.0.57004' from 'C:\Program Files (x86)\Microsoft Visual Studio\2019\Preview\MSBuild\Current\bin'. Use option -MSBuildVersion to force nuget to use a specific version of MSBuild.
C:\Program Files (x86)\Microsoft Visual Studio\2019\Preview\MSBuild\Current\bin\msbuild.exe "C:\Users\zivkan\AppData\Local\Temp\NuGetScratch\ucjxzxe1.nws.nugetinputs.targets" /t:GenerateRestoreGraphFile /nologo /nr:false /v:q /p:NuGetRestoreTargets="C:\Users\zivkan\AppData\Local\Temp\NuGetScratch\uu2pggtg.bpx.nugetrestore.targets" /p:RestoreUseCustomAfterTargets="True" /p:RestoreTaskAssemblyFile="C:\src\NuGet.Client\artifacts\VS15\NuGet.exe"

Running restore with 16 concurrent jobs.
Reading project file C:\src\NuGet.Client\build\bootstrap.proj.
The restore inputs for 'bootstrap' have not changed. No further actions are required to complete the restore.
Not all packages are on disk for: bootstrap
Restoring packages for C:\src\NuGet.Client\build\bootstrap.proj...
Restoring packages for .NETStandard,Version=v2.0...
Resolving conflicts for .NETStandard,Version=v2.0...
  CACHE https://api.nuget.org/v3-flatcontainer/newtonsoft.json/index.json
  CACHE https://dotnetfeed.blob.core.windows.net/dotnet-core/flatcontainer/newtonsoft.json/index.json
  GET https://pkgs.dev.azure.com/azure-public/3ccf6661-f8ce-4e8a-bb2e-eff943ddd3c7/_packaging/78665e4c-b767-412b-9804-2b1ef7a48b8a/nuget/v3/flat2/newtonsoft.json/index.json
  GET https://pkgs.dev.azure.com/azure-public/3ccf6661-f8ce-4e8a-bb2e-eff943ddd3c7/_packaging/2a239fd0-3e21-40b0-b9d6-bc122fec7eb2/nuget/v3/flat2/newtonsoft.json/index.json
  GET https://pkgs.dev.azure.com/dnceng/9ee6d478-d288-47f7-aacc-f6e6d082ae6d/_packaging/1a5f89f6-d8da-4080-b15f-242650c914a8/nuget/v3/flat2/newtonsoft.json/index.json
  GET https://pkgs.dev.azure.com/dnceng/9ee6d478-d288-47f7-aacc-f6e6d082ae6d/_packaging/40ffd17d-412e-4e95-b1ca-c5e6b3774cd2@dc1a9d3d-0b54-4674-a4c3-0f6e4269f5f4/nuget/v3/flat2/newtonsoft.json/index.json
  GET https://pkgs.dev.azure.com/dnceng/9ee6d478-d288-47f7-aacc-f6e6d082ae6d/_packaging/9d15d80a-6afc-4f7e-901b-9378146a4b8b/nuget/v3/flat2/newtonsoft.json/index.json
Acquiring lock for the installation of Newtonsoft.Json 9.0.1
Acquired lock for the installation of Newtonsoft.Json 9.0.1
  CACHE https://api.nuget.org/v3-flatcontainer/newtonsoft.json/9.0.1/newtonsoft.json.9.0.1.nupkg
  NotFound https://pkgs.dev.azure.com/azure-public/3ccf6661-f8ce-4e8a-bb2e-eff943ddd3c7/_packaging/78665e4c-b767-412b-9804-2b1ef7a48b8a/nuget/v3/flat2/newtonsoft.json/index.json 79ms
  NotFound https://pkgs.dev.azure.com/azure-public/3ccf6661-f8ce-4e8a-bb2e-eff943ddd3c7/_packaging/2a239fd0-3e21-40b0-b9d6-bc122fec7eb2/nuget/v3/flat2/newtonsoft.json/index.json 79ms
  NotFound https://pkgs.dev.azure.com/dnceng/9ee6d478-d288-47f7-aacc-f6e6d082ae6d/_packaging/9d15d80a-6afc-4f7e-901b-9378146a4b8b/nuget/v3/flat2/newtonsoft.json/index.json 93ms
  NotFound https://pkgs.dev.azure.com/dnceng/9ee6d478-d288-47f7-aacc-f6e6d082ae6d/_packaging/1a5f89f6-d8da-4080-b15f-242650c914a8/nuget/v3/flat2/newtonsoft.json/index.json 114ms
  NotFound https://pkgs.dev.azure.com/dnceng/9ee6d478-d288-47f7-aacc-f6e6d082ae6d/_packaging/40ffd17d-412e-4e95-b1ca-c5e6b3774cd2@dc1a9d3d-0b54-4674-a4c3-0f6e4269f5f4/nuget/v3/flat2/newtonsoft.json/index.json 115ms
PackageSignatureVerificationLog: PackageIdentity: Newtonsoft.Json.9.0.1 Source: https://api.nuget.org/v3/index.json PackageSignatureValidity: True
Installed Newtonsoft.Json 9.0.1 from https://api.nuget.org/v3/index.json with content hash U82mHQSKaIk+lpSVCbWYKNavmNH1i5xrExDEquU1i6I5pV6UMOqRnJRSlKO3cMPfcpp0RgDY+8jUXHdQ4IfXvw==.
Checking compatibility of packages on .NETStandard,Version=v2.0.
Checking compatibility for bootstrap 1.0.0 with .NETStandard,Version=v2.0.
Checking compatibility for NETStandard.Library 2.0.3 with .NETStandard,Version=v2.0.
Checking compatibility for Microsoft.NETCore.Platforms 1.1.0 with .NETStandard,Version=v2.0.
All packages and projects are compatible with .NETStandard,Version=v2.0.
Committing restore...
Generating MSBuild file C:\src\NuGet.Client\build\../packages/obj/bootstrap.proj.nuget.g.props.
Assets file has not changed. Skipping assets file writing. Path: C:\src\NuGet.Client\packages\obj\project.assets.json
Writing cache file to disk. Path: C:\src\NuGet.Client\packages\obj\project.nuget.cache
Persisting dg to C:\src\NuGet.Client\build\../packages/obj/bootstrap.proj.nuget.dgspec.json
Restored C:\src\NuGet.Client\build\bootstrap.proj (in 852 ms).

NuGet Config files used:
    C:\src\NuGet.Client\NuGet.Config
    C:\Users\zivkan\AppData\Roaming\NuGet\NuGet.Config
    C:\Program Files (x86)\NuGet\Config\Microsoft.VisualStudio.FallbackLocation.config
    C:\Program Files (x86)\NuGet\Config\Microsoft.VisualStudio.Offline.config

Feeds used:
    https://api.nuget.org/v3/index.json
    https://dotnetfeed.blob.core.windows.net/dotnet-core/index.json
    https://pkgs.dev.azure.com/azure-public/vside/_packaging/vs-impl/nuget/v3/index.json
    https://pkgs.dev.azure.com/azure-public/vside/_packaging/vssdk/nuget/v3/index.json
    https://pkgs.dev.azure.com/dnceng/public/_packaging/dotnet-eng/nuget/v3/index.json
    https://pkgs.dev.azure.com/dnceng/public/_packaging/myget-legacy%40Local/nuget/v3/index.json
    https://pkgs.dev.azure.com/dnceng/public/_packaging/nuget-build/nuget/v3/index.json
```

</details>

<details><summary>NuGet 5.8</summary>

```text
NuGet Version: 5.8.0.6930
MSBuild P2P timeout [ms]: 120000
MSBuild auto-detection: using msbuild version '16.9.0.57004' from 'C:\Program Files (x86)\Microsoft Visual Studio\2019\Preview\MSBuild\Current\bin'. Use option -MSBuildVersion to force nuget to use a specific version of MSBuild.
C:\Program Files (x86)\Microsoft Visual Studio\2019\Preview\MSBuild\Current\bin\msbuild.exe "C:\Users\zivkan\AppData\Local\Temp\NuGetScratch\tdpco0fq.qyg.nugetinputs.targets" /t:GenerateRestoreGraphFile /nologo /nr:false /v:q /p:NuGetRestoreTargets="C:\Users\zivkan\AppData\Local\Temp\NuGetScratch\xxra5iqe.axr.nugetrestore.targets" /p:RestoreUseCustomAfterTargets="True" /p:RestoreTaskAssemblyFile="C:\Users\zivkan\Downloads\bin\nuget.5.8.0.exe"

Running restore with 16 concurrent jobs.
Reading project file C:\src\NuGet.Client\build\bootstrap.proj.
The restore inputs for 'bootstrap' have not changed. No further actions are required to complete the restore.
Not all packages are on disk for: bootstrap
Restoring packages for C:\src\NuGet.Client\build\bootstrap.proj...
Restoring packages for .NETStandard,Version=v2.0...
Resolving conflicts for .NETStandard,Version=v2.0...
  CACHE https://api.nuget.org/v3-flatcontainer/newtonsoft.json/index.json
  CACHE https://dotnetfeed.blob.core.windows.net/dotnet-core/flatcontainer/newtonsoft.json/index.json
  GET https://pkgs.dev.azure.com/azure-public/3ccf6661-f8ce-4e8a-bb2e-eff943ddd3c7/_packaging/78665e4c-b767-412b-9804-2b1ef7a48b8a/nuget/v3/flat2/newtonsoft.json/index.json
  GET https://pkgs.dev.azure.com/azure-public/3ccf6661-f8ce-4e8a-bb2e-eff943ddd3c7/_packaging/2a239fd0-3e21-40b0-b9d6-bc122fec7eb2/nuget/v3/flat2/newtonsoft.json/index.json
  GET https://pkgs.dev.azure.com/dnceng/9ee6d478-d288-47f7-aacc-f6e6d082ae6d/_packaging/1a5f89f6-d8da-4080-b15f-242650c914a8/nuget/v3/flat2/newtonsoft.json/index.json
  GET https://pkgs.dev.azure.com/dnceng/9ee6d478-d288-47f7-aacc-f6e6d082ae6d/_packaging/40ffd17d-412e-4e95-b1ca-c5e6b3774cd2@dc1a9d3d-0b54-4674-a4c3-0f6e4269f5f4/nuget/v3/flat2/newtonsoft.json/index.json
  GET https://pkgs.dev.azure.com/dnceng/9ee6d478-d288-47f7-aacc-f6e6d082ae6d/_packaging/9d15d80a-6afc-4f7e-901b-9378146a4b8b/nuget/v3/flat2/newtonsoft.json/index.json
Acquiring lock for the installation of Newtonsoft.Json 9.0.1
Acquired lock for the installation of Newtonsoft.Json 9.0.1
Installing Newtonsoft.Json 9.0.1.
  CACHE https://api.nuget.org/v3-flatcontainer/newtonsoft.json/9.0.1/newtonsoft.json.9.0.1.nupkg
  NotFound https://pkgs.dev.azure.com/azure-public/3ccf6661-f8ce-4e8a-bb2e-eff943ddd3c7/_packaging/2a239fd0-3e21-40b0-b9d6-bc122fec7eb2/nuget/v3/flat2/newtonsoft.json/index.json 59ms
  NotFound https://pkgs.dev.azure.com/dnceng/9ee6d478-d288-47f7-aacc-f6e6d082ae6d/_packaging/1a5f89f6-d8da-4080-b15f-242650c914a8/nuget/v3/flat2/newtonsoft.json/index.json 83ms
  NotFound https://pkgs.dev.azure.com/dnceng/9ee6d478-d288-47f7-aacc-f6e6d082ae6d/_packaging/9d15d80a-6afc-4f7e-901b-9378146a4b8b/nuget/v3/flat2/newtonsoft.json/index.json 93ms
  NotFound https://pkgs.dev.azure.com/dnceng/9ee6d478-d288-47f7-aacc-f6e6d082ae6d/_packaging/40ffd17d-412e-4e95-b1ca-c5e6b3774cd2@dc1a9d3d-0b54-4674-a4c3-0f6e4269f5f4/nuget/v3/flat2/newtonsoft.json/index.json 103ms
  NotFound https://pkgs.dev.azure.com/azure-public/3ccf6661-f8ce-4e8a-bb2e-eff943ddd3c7/_packaging/78665e4c-b767-412b-9804-2b1ef7a48b8a/nuget/v3/flat2/newtonsoft.json/index.json 266ms
PackageSignatureVerificationLog: PackageIdentity: Newtonsoft.Json.9.0.1 Source: https://api.nuget.org/v3/index.json PackageSignatureValidity: True
Completed installation of Newtonsoft.Json 9.0.1
Checking compatibility of packages on .NETStandard,Version=v2.0.
Checking compatibility for bootstrap 1.0.0 with .NETStandard,Version=v2.0.
Checking compatibility for NETStandard.Library 2.0.3 with .NETStandard,Version=v2.0.
Checking compatibility for Microsoft.NETCore.Platforms 1.1.0 with .NETStandard,Version=v2.0.
All packages and projects are compatible with .NETStandard,Version=v2.0.
Committing restore...
Generating MSBuild file C:\src\NuGet.Client\build\../packages/obj/bootstrap.proj.nuget.g.props.
Assets file has not changed. Skipping assets file writing. Path: C:\src\NuGet.Client\packages\obj\project.assets.json
Writing cache file to disk. Path: C:\src\NuGet.Client\packages\obj\project.nuget.cache
Persisting dg to C:\src\NuGet.Client\build\../packages/obj/bootstrap.proj.nuget.dgspec.json
Restored C:\src\NuGet.Client\build\bootstrap.proj (in 1.05 sec).

NuGet Config files used:
    C:\src\NuGet.Client\NuGet.Config
    C:\Users\zivkan\AppData\Roaming\NuGet\NuGet.Config
    C:\Program Files (x86)\NuGet\Config\Microsoft.VisualStudio.FallbackLocation.config
    C:\Program Files (x86)\NuGet\Config\Microsoft.VisualStudio.Offline.config

Feeds used:
    https://api.nuget.org/v3/index.json
    https://dotnetfeed.blob.core.windows.net/dotnet-core/index.json
    https://pkgs.dev.azure.com/azure-public/vside/_packaging/vs-impl/nuget/v3/index.json
    https://pkgs.dev.azure.com/azure-public/vside/_packaging/vssdk/nuget/v3/index.json
    https://pkgs.dev.azure.com/dnceng/public/_packaging/dotnet-eng/nuget/v3/index.json
    https://pkgs.dev.azure.com/dnceng/public/_packaging/myget-legacy%40Local/nuget/v3/index.json
    https://pkgs.dev.azure.com/dnceng/public/_packaging/nuget-build/nuget/v3/index.json
```

</details>

Just installing/installed messages:

NuGet 5.8:

```text
Installing Newtonsoft.Json 9.0.1.
Completed installation of Newtonsoft.Json 9.0.1
```

With this PR:
```text
Installed Newtonsoft.Json 9.0.1 from https://dotnetfeed.blob.core.windows.net/dotnet-core/index.json with content hash U82mHQSKaIk+lpSVCbWYKNavmNH1i5xrExDEquU1i6I5pV6UMOqRnJRSlKO3cMPfcpp0RgDY+8jUXHdQ4IfXvw==.
```

File: src/NuGet.Core/NuGet.Packaging.Extraction/NuGet.Packaging.Extraction.csproj (hunk 1 above)
Comment (zivkan) [reply src/NuGet.Core/NuGet.Packaging.Extraction/NuGet.Packaging.Extraction.csproj]: typo, although it's hard to guess what I was going for. 😕

File: src/NuGet.Core/NuGet.Packaging/PackageExtractor.cs (hunk 2 above)
Comment (zivkan) [reply src/NuGet.Core/NuGet.Packaging/PackageExtractor.cs]: Please see the new comment: https://github.com/NuGet/NuGet.Client/pull/3849#issuecomment-765764223

File: src/NuGet.Core/NuGet.Packaging.Extraction/Strings.resx [hunk 3]
```diff
@@ -138,10 +138,12 @@
   <data name="ErrorNullOrEmptyPackageId" xml:space="preserve">
     <value>Null or empty package id</value>
   </data>
-  <data name="Log_InstallingPackage" xml:space="preserve">
-    <value>Installing {0} {1}.</value>
+  <data name="Log_InstalledPackage" xml:space="preserve">
```
Comment (nkolev92) [inline src/NuGet.Core/NuGet.Packaging.Extraction/Strings.resx:141]: I know there's been a chat about this, can you please remind why we have to duplicate this file and instead just embed it?

**Review by nkolev92 — APPROVED** [review]

File: src/NuGet.Core/NuGet.Packaging.Extraction/Strings.resx (hunk 3 above)
Comment (zivkan) [reply src/NuGet.Core/NuGet.Packaging.Extraction/Strings.resx:141]: When I linked `Strings.resx`, but also keeping the `Generator` metadata, the resx code generator creates a designer.cs file where the namespace is `NuGet.Packaging.Extraction...NuGet.Packaging`, which fails to compile. When I don't set the generator, and compare the binlog, I see `Strings.resx`'s `OutputResource` metadata is a little different, and therefore the command line arguments for `csc.exe` are a little different. It looks like it should work, it certainly compiles. But I'm not sure it'll actually work at runtime, and I'm not willing to spend more time investigating.

**Pull request merged**